$OpenBSD: patch-frontends_php_popup_bitem_php,v 1.1 2012/08/05 19:29:35 jasper Exp $

Security fix for CVE-2012-3435
'itemid' was not properly sanitized which would lead to an SQL injection flaw.

--- frontends/php/popup_bitem.php.orig	Sun Jul 29 22:50:31 2012
+++ frontends/php/popup_bitem.php	Sun Jul 29 22:52:46 2012
@@ -60,7 +60,7 @@ include_once 'include/page_header.php';
 
 	$_REQUEST['caption'] = get_request('caption','');
 	$_REQUEST['axisside'] = get_request('axisside',	GRAPH_YAXIS_SIDE_LEFT);
-	if(zbx_empty($_REQUEST['caption']) && isset($_REQUEST['itemid']) && ($_REQUEST['itemid'] > 0)){
+	if(zbx_empty($_REQUEST['caption']) && sset($_REQUEST['itemid']) && $_REQUEST['itemid'] && zbx_is_int($_REQUEST['itemid'])){
 		$_REQUEST['caption'] = item_description(get_item_by_itemid($_REQUEST['itemid']));
 	}
 
@@ -108,7 +108,7 @@ include_once 'include/page_header.php';
 		$axisside	= get_request('axisside',	GRAPH_YAXIS_SIDE_LEFT);
 
 		$description = '';
-		if($itemid > 0){
+		if($itemid > 0 && zbx_is_int($itemid)){
 			$description = get_item_by_itemid($itemid);
 			$description = item_description($description);
 		}
